### ISSUE TITLE
Use endpoint information (when available) over request url for profiler name on non MVC/Razor routes.

### ISSUE THREAD

**Issue by stevebelskie** [MiniProfiler/dotnet#514]
Currently, for requests that are not either controller actions or razor pages, the request url is used for the profiler name. Now that endpoint routing is a thing, the `DisplayName` property provides a better option in cases where it exists. 

```cs
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllerRoute(
                    name: "default",
                    pattern: "{controller=Home}/{action=Index}/{id?}");

                endpoints.MapGet("get-versions", async context => await context.Response.WriteAsync(/**some value**/); )
                         .Add(builder =>
                         {
                             builder.DisplayName = "Get Versions";
                         });
            });
```

Using "Get Versions" instead of "https://domain/get-versions" as the name for the non-MVC endpoint here seems like an improvement and would greatly reduce the ugliness that comes with overlong urls being chopped at 50 characters when used for the profiler name.

From a cursory look at the code, I think the `EnsureName` method in the middleware could be adjusted to include another conditional, but I could be missing something. Something along the lines of this:

```cs
                var routeData = context.GetRouteData();
                Endpoint endpoint;
                if (routeData?.Values["controller"] != null)
                {
                    profiler.Name = routeData.Values["controller"] + "/" + routeData.Values["action"];
                }
                else if (routeData?.Values["page"] != null)
                {
                    profiler.Name = routeData.Values["page"].ToString();
                }
                else if((endpoint = context.GetEndpoint() != null))
                {
                    profiler.Name = enpoint.DisplayName;
                }
                else
                {
                    profiler.Name = url;
                    if (profiler.Name.Length > 50)
                        profiler.Name = profiler.Name.Remove(50);
                }
```

**Comment by NickCraver** [comment]
Good call! Yeah let's add something like (latest C# allows shorter syntax):
```cs
else if (context.GetEndpoint() is Endpoint endPoint && endPoint.DisplayName.HasValue())
{
    profiler.Name = endPoint.DisplayName;
}
```

I'll try and repro this soon as time allows, or a PR adding an example and fix to the samples project would also be most welcome!

**Comment by stevebelskie** [comment]
How does something like #517 seem to you?

**Comment by NickCraver** [comment]
Implemented in #517, thank you for this!
